### PR TITLE
Remove YAML from HSE

### PR DESCRIPTION
**Pull request by tristan957**
Removes hse_util/yaml and libyaml as a dependency. We have decided to
plow forward with JSON, and libyaml was barely even used within HSE
anywhere. I changed the CLI to produce JSON where it was previously YAML
and I changed merge_test to use JSON.

Signed-off-by: Tristan Partin <tpartin@micron.com>

### CODE REVIEW

**Comment by tristan957** [comment]
Not going to merge until the parent PR is merged.

**Comment by alexttx** [comment]
I don't think the CLI should produce JSON by default.  I think we used YAML because it was both structured and human readable.

**Comment by tristan957** [comment]
Yea I don't think so either. Merely a place holder for the PR until someone proposes better options though it has been a while since I looked at this PR and don't really plan to come back to it until the REST PR merges.

File: tests/unit/cn/merge-test-cases/generated/crafted-cross_kvset_dups1.json [hunk 1]
```diff
@@ -0,0 +1,128 @@
+{
```
Comment (alexttx) [inline tests/unit/cn/merge-test-cases/generated/crafted-cross_kvset_dups1.json:1]: These files are generated by `make-merge-test-cases.py`.  Should update that script to produce JSON.

Comment (tristan957) [reply tests/unit/cn/merge-test-cases/generated/crafted-cross_kvset_dups1.json:1]: Done, though that script and what it generates needs a lot more love. A task for another day, I guess

**Review by alexttx — APPROVED** [review]

**Comment by alexttx** [comment]
I'm running on master with this PR + a few others merged locally and ran into the following issue.  Not sure if it a new issue with this PR or master.

```
[0 deepblue ~]$ hse kvdb info $db
AddressSanitizer:DEADLYSIGNAL
=================================================================
==2634008==ERROR: AddressSanitizer: SEGV on unknown address 0x000000000000 (pc 0x7f6f4a6fbf3d bp 0x7ffd37c00300 sp 0x7ffd37bffaa8 T0)
==2634008==The signal is caused by a READ memory access.
==2634008==Hint: address points to the zero page.
    #0 0x7f6f4a6fbf3d in __strlen_avx2_rtm (/lib64/libc.so.6+0x18df3d)
    #1 0x7f6f4b01bb1c in __interceptor_puts.part.0 (/lib64/libasan.so.6+0x3fb1c)
    #2 0x40435e in kvdb_info_props ../../../../home/atomlinson/w/src/hse/hse/cli/cli_util.c:76
    #3 0x404614 in kvdb_info_print ../../../../home/atomlinson/w/src/hse/hse/cli/cli_util.c:98
    #4 0x4094c8 in cli_hse_kvdb_info_impl ../../../../home/atomlinson/w/src/hse/hse/cli/hse_cli.c:680
    #5 0x40a8cc in cli_hse_kvdb_info ../../../../home/atomlinson/w/src/hse/hse/cli/hse_cli.c:932
    #6 0x409aee in cli_hse_kvdb ../../../../home/atomlinson/w/src/hse/hse/cli/hse_cli.c:771
    #7 0x40fcf4 in cli_hse ../../../../home/atomlinson/w/src/hse/hse/cli/hse_cli.c:1872
    #8 0x40ff76 in main ../../../../home/atomlinson/w/src/hse/hse/cli/hse_cli.c:1892
    #9 0x7f6f4a59b55f in __libc_start_call_main (/lib64/libc.so.6+0x2d55f)
    #10 0x7f6f4a59b60b in __libc_start_main_alias_1 (/lib64/libc.so.6+0x2d60b)
    #11 0x4038e4 in _start (/opt/hse/bin/hse3+0x4038e4)

AddressSanitizer can not provide additional info.
SUMMARY: AddressSanitizer: SEGV (/lib64/libc.so.6+0x18df3d) in __strlen_avx2_rtm
==2634008==ABORTING
```

**Comment by tristan957** [comment]
Probably with this PR. Will look into it. Thanks for the report.

File: cli/cli_util.c [hunk 2]
```diff
@@ -301,16 +245,21 @@ kvdb_compact_request(const char *kvdb_home, const char *request_type, u32 timeou
...
-
-        printf("%s", stat_buf);
+        printf(
+            "samp_lwm: %d\n"
+            "samp_hwm: %d\n"
+            "samp_curr: %d\n"
+            "active: %s\n"
+            "canceled: %s\n",
+            status.kvcs_samp_lwm, status.kvcs_samp_hwm, status.kvcs_samp_curr,
+            status.kvcs_active ? "true" : "false", status.kvcs_canceled ? "true" : "false");
```
Comment (beckerg) [inline cli/cli_util.c:255]: didn't this used to print in floating point?

File: cli/lib/rest/api.c [hunk 3]
```diff
@@ -319,30 +319,30 @@ kvdb_get_kvs_names_cb(
...
-    buf = calloc(len, sizeof(*namev) + HSE_KVS_NAME_LEN_MAX);
-    if (!buf) {
+    namev = calloc(len, sizeof(*namev) + HSE_KVS_NAME_LEN_MAX);
+    if (!namev) {
         err = merr(ENOMEM);
         goto out;
     }
 
-    namev = &buf;
+    name = (char *)&namev[len];
```
Comment (beckerg) [inline cli/lib/rest/api.c:328]: confusing..   isn't  'name = namev[len];'  the same thing?

Comment (tristan957) [reply cli/lib/rest/api.c:328]: Maybe. Let me test. Pulled this out of the kvs_names logic in ikvdb.c

File: cli/cli_util.c (hunk 2 above)
Comment (tristan957) [reply cli/cli_util.c:255]: Yes, would you like me to revert back?

File: cli/lib/rest/api.c (hunk 3 above)
Comment (beckerg) [reply cli/lib/rest/api.c:328]: nevermind..

Comment (beckerg) [reply cli/lib/rest/api.c:328]: ah, i would prob write this as  'name = (char *)(name + len);', but either or...

Comment (tristan957) [reply cli/lib/rest/api.c:328]: apparently it isn't the same thing

**Review by beckerg — APPROVED** [review]

File: cli/lib/rest/api.c (hunk 3 above)
Comment (beckerg) [reply cli/lib/rest/api.c:328]: baffling...  not sure what's going on there...

Comment (tristan957) [reply cli/lib/rest/api.c]: It is

[ptr ptr..., HSE_KVS_NAME_LEN_MAX HSE_KVS_NAME_LEN_MAX...]

**Pull request merged**